### PR TITLE
Use omitempty for providerConfigs

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

```release-note
none
```

### CODE REVIEW

File: pkg/cloudprovider/provider/aws/provider.go [hunk 1]
```diff
@@ -113,24 +113,24 @@ var (
...
 
 type RawConfig struct {
-	AccessKeyID     providerconfig.ConfigVarString `json:"accessKeyId"`
-	SecretAccessKey providerconfig.ConfigVarString `json:"secretAccessKey"`
+	AccessKeyID     providerconfig.ConfigVarString `json:"accessKeyId,omitempty"`
+	SecretAccessKey providerconfig.ConfigVarString `json:"secretAccessKey,omitempty"`
 
-	Region           providerconfig.ConfigVarString `json:"region"`
-	AvailabilityZone providerconfig.ConfigVarString `json:"availabilityZone"`
+	Region           providerconfig.ConfigVarString `json:"region,omitempty"`
```
Comment (mrIncompetent) [inline pkg/cloudprovider/provider/aws/provider.go]: According to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required only optional fields should use `omitempty`.

Comment (alvaroaleman) [reply pkg/cloudprovider/provider/aws/provider.go]: But none of these fields is required as they can be populated via the env

Comment (mrIncompetent) [reply pkg/cloudprovider/provider/aws/provider.go]: Only a subset of the fields are being fetched from the env.
Region is an example

Comment (alvaroaleman) [reply pkg/cloudprovider/provider/aws/provider.go]: But it could still come from a configmap or secret or not?

Comment (mrIncompetent) [reply pkg/cloudprovider/provider/aws/provider.go]: Which means the field is not "empty"

**Comment by alvaroaleman** [comment]
@mrIncompetent Updated, PTAL

**Pull request merged**